pkg/buildscript/buildscript.go

```diff
@@ -39,14 +60,90 @@ func (b *BuildScript) Equals(other *BuildScript) (bool, error) {
 }
 
 func (b *BuildScript) Clone() (*BuildScript, error) {
-	m, err := b.Marshal()
+	bb, err := deep.Copy(b)
 	if err != nil {
-		return nil, errs.Wrap(err, "unable to marshal this buildscript")
+		return nil, errs.Wrap(err, "unable to clone buildscript")
 	}
+	return bb, nil
+}
 
-	u, err := Unmarshal(m)
-	if err != nil {
-		return nil, errs.Wrap(err, "unable to unmarshal buildscript")
+// FuncCall is the exportable version of funcCall, because we do not want to expose low level buildscript functionality
```

Comment:
fwiw all the changes in this file are around making sense of buildscript data outside of the context of the buildscript package. We don't want to expose the raw data here, as that's considered too low level for external use.

Another way of putting it: if and when we refactor buildscripts (let's be real: when), we won't have to also refactor all use of the raw types. Instead there'll be a more limited set of APIs that will need to be addressed or refactored.